### PR TITLE
Tools imporvements and small fixes

### PR DESCRIPTION
**Pull request by melnikovalex**
Hey Ehsan,
there are several improvements for pyRevit Tools. Take a look, please

**Toggle imported**

- Activates Temporary view settings if the parameter is overwritten by View Template

**Selection tools (elements of the same type, the same family**

- Allow to preselect type from project browser (Symbol)

**Match/compare properties**

- Do not try to copy sheet number
- Compare views option

**Flip**

- Show only available flip option
- Rotate 180 for FamilyInstances which do not have FlipControl

**Tag in all views**

- Skip not placed Rooms/Spaces
- Select created tags after all

**Misk fixes**
 - `revit.query.get_linked_model_doc(extref)`
 - `revit.selection.pick_linked`(s)

### CODE REVIEW

File: extensions/pyRevitTools.extension/pyRevit.tab/Toggles.panel/toggles2.stack/Imported.pushbutton/script.py [hunk 1]
```diff
@@ -1,11 +1,26 @@
...
+    view_template = view.Document.GetElement(view.ViewTemplateId)
+    if DB.ElementId(int(built_in_parameter)) in view_template.GetNonControlledTemplateParameterIds():
+        return False
+    else:
+        return True
 
 @revit.carryout('Toggle Imported')
 def toggle_imported():
     aview = revit.active_view
+    # check if 'Imported' is overrided by Template. If so, enable temporary view
```
Comment (melnikovalex) [inline extensions/pyRevitTools.extension/pyRevit.tab/Toggles.panel/toggles2.stack/Imported.pushbutton/script.py:20]: I find this issue pretty critical, actually. At the moment, Revit API allows to change `VIS_GRAPHICS_IMPORT` parameter of a view even if it was overwritten by View Template. After that there is almost no way to discover the reason of the issue. And the only way to change it back after that is to disable Template and set it back (I am gonna write to Autodesk about this issue).
Here script activates Temporary view settings if the parameter is overwritten by View Template

File: extensions/pyRevitTools.extension/pyRevit.tab/Selection.panel/select.stack/Select.pulldown/Select All Elements Of Same Family.pushbutton/script.py [hunk 2]
```diff
@@ -13,7 +13,10 @@
 
 for el in selection:
     try:
-        family = el.Symbol.Family
+        if isinstance(el, DB.FamilySymbol):
```
Comment (melnikovalex) [inline extensions/pyRevitTools.extension/pyRevit.tab/Selection.panel/select.stack/Select.pulldown/Select All Elements Of Same Family.pushbutton/script.py:16]: Support Family Symbol as input. It allows user to select Family Type from project browser (the same below)

File: extensions/pyRevitTools.extension/pyRevit.tab/Modify.panel/edit1.stack/Match.splitpushbutton/Match Properties.pushbutton/script.py [hunk 3]
```diff
@@ -76,21 +76,34 @@ def get_source_properties(src_element):
 
     src_type = revit.query.get_type(src_element)
 
+    ignore_bi_parameters = []
+    # sheet number cannot be copied
+    if isinstance(src_element, DB.ViewSheet):
+        ignore_bi_parameters.append(int(DB.BuiltInParameter.SHEET_NUMBER))
```
Comment (melnikovalex) [inline extensions/pyRevitTools.extension/pyRevit.tab/Modify.panel/edit1.stack/Match.splitpushbutton/Match Properties.pushbutton/script.py]: Revit doesn't allow two Sheets to have the same number. Therefore it makes no sense to copy its value

File: extensions/pyRevitTools.extension/pyRevit.tab/Modify.panel/edit1.stack/Match.splitpushbutton/Compare Properties.pushbutton/script.py [hunk 4]
```diff
@@ -108,11 +108,29 @@ def compare_props(src_element, tgt_element):
                 )
 
 # main
+target_type = \
```
Comment (melnikovalex) [inline extensions/pyRevitTools.extension/pyRevit.tab/Modify.panel/edit1.stack/Match.splitpushbutton/Compare Properties.pushbutton/script.py]: Support views the same way it's done in "Match properties"

File: extensions/pyRevitTools.extension/pyRevit.tab/Modify.panel/edit2.stack/Flip.pushbutton/script.py [hunk 5]
```diff
@@ -1,25 +1,49 @@
 """Provides options for flipping (Hand/Face) selected elements."""
 
-from pyrevit import revit, DB
+from pyrevit import revit, DB, script
 from pyrevit import forms
 
 
 __context__ = 'selection'
 
+def can_flip_wall(el):
```
Comment (melnikovalex) [inline extensions/pyRevitTools.extension/pyRevit.tab/Modify.panel/edit2.stack/Flip.pushbutton/script.py:9]: I've added these small functions to check which options are relevant with current selection (see below)

File: extensions/pyRevitTools.extension/pyRevit.tab/Modify.panel/edit2.stack/Flip.pushbutton/script.py [hunk 6]
```diff
@@ -1,25 +1,49 @@
...
 def flip_facing():
     with revit.Transaction('Flip Facing Selected'):
         for el in revit.get_selection():
-            if hasattr(el, 'flipFacing'):
+            if hasattr(el, 'CanFlipFacing') and el.CanFlipFacing:
                 el.flipFacing()
-            elif hasattr(el, 'Flip'):
+            elif can_flip(el):
                 el.Flip()
+            elif hasattr(el, 'rotate'):
```
Comment (melnikovalex) [inline extensions/pyRevitTools.extension/pyRevit.tab/Modify.panel/edit2.stack/Flip.pushbutton/script.py:39]: If FlipFacing is not available - rotate the family to 180 Degrees (works the same for Doors, Windows, etc.)

File: extensions/pyRevitTools.extension/pyRevit.tab/Modify.panel/edit2.stack/Flip.pushbutton/script.py [hunk 7]
```diff
@@ -59,16 +83,32 @@ def flip_wall_location_line():
                     return
 
 
+def selection_can(func):
```
Comment (melnikovalex) [inline extensions/pyRevitTools.extension/pyRevit.tab/Modify.panel/edit2.stack/Flip.pushbutton/script.py:86]: Checks if selected elements answer certain requirements

File: extensions/pyRevitTools.extension/pyRevit.tab/Modify.panel/edit2.stack/Flip.pushbutton/script.py [hunk 8]
```diff
@@ -59,16 +83,32 @@ def flip_wall_location_line():
...
+    return False
+
+
+options_dict = {}
 
-options_dict = {'Flip Facing': flip_facing,
-                'Flip Hand': flip_hand,
-                'Flip Wall On CenterLine': flip_wall_location,
-                'Flip Wall Location Line': flip_wall_location_line}
+if selection_can(can_flip_wall):
```
Comment (melnikovalex) [inline extensions/pyRevitTools.extension/pyRevit.tab/Modify.panel/edit2.stack/Flip.pushbutton/script.py:95]: Show certain options depending on selected elements

File: extensions/pyRevitTools.extension/pyRevit.tab/Modify.panel/edit3.stack/Edit.pulldown/Tag All in All Views.pushbutton/script.py [hunk 9]
```diff
@@ -16,18 +16,19 @@
 
 
 def GetElementCenter(el):
-    cen = el.Location.Point
-    z = (el.UpperLimit.Elevation + el.LimitOffset) / 2
-    cen = cen.Add(DB.XYZ(0, 0, z))
-    return cen
+    if el.Location:
```
Comment (melnikovalex) [inline extensions/pyRevitTools.extension/pyRevit.tab/Modify.panel/edit3.stack/Edit.pulldown/Tag All in All Views.pushbutton/script.py:19]: Skip unlocated rooms/spaces

File: extensions/pyRevitTools.extension/pyRevit.tab/Modify.panel/edit3.stack/Edit.pulldown/Tag All in All Views.pushbutton/script.py [hunk 10]
```diff
@@ -85,4 +90,6 @@ def tag_all_spaces():
 option_func = options_dict.get(selected_switch, None)
 
 if option_func:
-    option_func()
+    tag_ids = option_func()
```
Comment (melnikovalex) [inline extensions/pyRevitTools.extension/pyRevit.tab/Modify.panel/edit3.stack/Edit.pulldown/Tag All in All Views.pushbutton/script.py:93]: Selects created tags so user can change their types

File: extensions/pyRevitTools.extension/pyRevit.tab/Toggles.panel/toggles2.stack/Imported.pushbutton/script.py (hunk 1 above)
Comment (melnikovalex) [reply extensions/pyRevitTools.extension/pyRevit.tab/Toggles.panel/toggles2.stack/Imported.pushbutton/script.py:20]: Just for info, there is an Autodesk ticket with number `REVIT-153828` for the issue

File: pyrevitlib/pyrevit/revit/db/query.py [hunk 11]
```diff
@@ -524,7 +525,7 @@ def get_linked_model_doc(linked_model):
 
     if lmodel:
         for open_doc in HOST_APP.docs:
-            if open_doc.Title == lmodel.name:
+            if os.path.split(open_doc.PathName)[1] == lmodel.name:
```
Comment (melnikovalex) [inline pyrevitlib/pyrevit/revit/db/query.py:562]: open_doc.Title did return filename without extension, and lmodel.name does with extenstion

**Comment by eirannejad** [comment]
Merged to `pr/714` for review

**Pull request merged**